src/main/java/com/denizenscript/depenizen/bukkit/properties/auraskills/AuraSkillsPlayerExtensions.java

```diff
@@ -0,0 +1,120 @@
+        // - narrate <player.auraskills_levels>
+        // -->
+        PlayerTag.tagProcessor.registerTag(MapTag.class,  "auraskills_levels", (attribute, player) -> {
+            MapTag levels = new MapTag();
+            SkillsUser user = AuraSkillsApi.get().getUser(player.getUUID());
+            for (Skills skill : Skills.values()) {
+                levels.putObject(skill.name(), new ElementTag(user.getSkillLevel(skill)));
+            }
+            return levels;
+        });
+
+        // <--[tag]
+        // @attribute <PlayerTag.auraskills_experience>
+        // @returns MapTag
+        // @plugin Depenizen, AuraSkills
+        // @mechanism PlayerTag.auraskills_experience
+        // @description
+        // Returns a map of a player's skill experience in the format "<skill>=<experience>".
+        // @example
+        // # Returns 'map@[FARMING=0;FORAGING=0;MINING=0;FISHING=0;EXCAVATION=0;ARCHERY=0;FIGHTING=0;DEFENSE=0;AGILITY=0;ENDURANCE=0;ALCHEMY=0;ENCHANTING=0;SORCERY=0;HEALING=0;FORGING=0]' on a brand new player.
+        // - narrate <player.auraskills_experience>
+        // -->
+        PlayerTag.tagProcessor.registerTag(MapTag.class,  "auraskills_experience", (attribute, player) -> {
+            MapTag levels = new MapTag();
+            SkillsUser user = AuraSkillsApi.get().getUser(player.getUUID());
+            for (Skills skill : Skills.values()) {
+                levels.putObject(skill.name(), new ElementTag(user.getSkillXp(skill)));
+            }
+            return levels;
+        });
+
+        // <--[mechanism]
+        // @object PlayerTag
+        // @name auraskills_levels
+        // @input MapTag
+        // @plugin Depenizen, AuraSkills
+        // @description
+        // Changes a player's skill level. Inputted as "<skill>=<level>".
+        // @tags
+        // <PlayerTag.auraskills_levels>
+        // @example
+        // # Sets the player's agility level to 5 and their excavation level to 3.
+        // - adjust <player> auraskills_levels:agility=5;excavation=3
+        // -->
+        PlayerTag.registerOnlineOnlyMechanism("auraskills_levels", MapTag.class, (player, mechanism, value) -> {
+            boolean valid = false;
+            SkillsUser user = AuraSkillsApi.get().getUser(player.getUUID());
+            for (Skills skill : Skills.values()) {
+                if (value.containsKey(skill.name().toLowerCase())) {
+                    ElementTag experience = value.getElement(skill.name());
+                    if (!experience.isInt() || experience.asInt() < 0) {
+                        mechanism.echoError("'" + experience + "' is not a valid level.");
+                    }
+                    else {
+                        user.setSkillLevel(skill, experience.asInt());
+                    }
+                    valid = true;
+                }
+            }
+            if (!valid) {
+                mechanism.echoError("There are no valid skills as part of the input.");
+            }
```

Comment:
This is still not implemented as monkey suggested earlier - should loop over the map's entries and get the skill from each key instead of looping over all the skills and checking if they're in the map.
Would also allow having more specific error messages if a key in the map is invalid.